shards/modules/imaging/imaging.cpp

```diff
@@ -308,11 +312,115 @@ struct Resize {
+
+  void cleanup() { PARAM_CLEANUP(); }
+
+  auto static constexpr Conv_UIntToFloat4 = [](const SHImage &image, auto *pixel) {
+    constexpr float max = float(std::numeric_limits<std::remove_pointer_t<decltype(pixel)>>::max());
+    Float4VarPayload result{};
+    for (size_t i = 0; i < image.channels; i++)
+      result.float4Value[i] = (float)pixel[i] / max;
+    return result;
+  };
+
+  auto static constexpr Conv_FloatToInt4 = [](const SHImage &image, auto *pixel) {
+    constexpr float max = float(std::numeric_limits<std::remove_pointer_t<decltype(pixel)>>::max());
+    Int4VarPayload result{};
+    for (size_t i = 0; i < image.channels; i++)
+      result.int4Value[i] = (int32_t)pixel[i] * max;
+    return result;
+  };
+
+  auto static constexpr Conv_CastFloat4 = [](const SHImage &image, auto *pixel) {
+    Float4VarPayload result{};
+    for (size_t i = 0; i < image.channels; i++)
+      result.float4Value[i] = (float)pixel[i];
+    return result;
+  };
+
+  auto static constexpr Conv_CastInt4 = [](const SHImage &image, auto *pixel) {
+    Int4VarPayload result{};
+    for (size_t i = 0; i < image.channels; i++)
+      result.int4Value[i] = (int32_t)pixel[i];
+    return result;
+  };
+
+  template <typename TPixel, typename TConv> SHVar convert(const SHImage &image, SHInt2 coord, TConv conv) {
+    TPixel *pix = (TPixel *)image.data + image.channels * (coord[1] * image.width + coord[0]);
+    return Var(conv(image, pix));
+  }
+
+  SHVar activate(SHContext *context, const SHVar &input) {
+    auto &image = _image.get().payload.imageValue;
+    int w = uint32_t(image.width);
+    int h = uint32_t(image.height);
+
+    SHInt2 coord = input.payload.int2Value;
+    if (coord[0] < 0 || coord[0] >= w) {
+      if (!_default->isNone())
+        return _default;
+      throw std::out_of_range("Image fetch x coordinate out of range");
+    }
+    if (coord[1] < 0 || coord[1] >= h) {
+      if (!_default->isNone())
+        return _default;
+      throw std::out_of_range("Image fetch y coordinate out of range");
+    }
+
+    auto pixsize = getPixelSize(_image.get());
+
+    if (pixsize == 1) {
+      return (bool)*_asInteger ? convert<uint8_t>(image, coord, Conv_CastInt4) //
+                              : convert<uint8_t>(image, coord, Conv_UIntToFloat4);
+    } else if (pixsize == 2) {
+      return (bool)*_asInteger ? convert<uint16_t>(image, coord, Conv_CastInt4) //
+                              : convert<uint16_t>(image, coord, Conv_UIntToFloat4);
+    } else if (pixsize == 4) {
+      return (bool)*_asInteger ? convert<float>(image, coord, Conv_FloatToInt4) //
+                              : convert<float>(image, coord, Conv_CastFloat4);
+    } else {
+      throw std::logic_error("Invalid image format");
+    }
+  }
+};
+
 } // namespace Imaging
 } // namespace shards
 
 SHARDS_REGISTER_FN(imaging) {
   using namespace shards::Imaging;
+
+  REGISTER_SHARD("GetImagePixel", ImageGetPixel);
```

Comment:
We never mention Image, lets call it `GetPixel`, or actually `TakePixel`

Reply:
#730